### ISSUE TITLE
Modernizr.csstransitions issue with older Webkit versions 2.8.3

### ISSUE THREAD

**Issue by stnor** [Modernizr/Modernizr#1501]
I get a false positive from Modernizr.csstransitions with versions of Webkit below 537 as they seem to NOT support css transitions (Android, ipad, iphone error repors from my users).

I added the following code to work around the issue;

```javascript
      if (!this.isBrokenWebkitBrowser() && Modernizr.csstransitions) ....

       isBrokenWebkitBrowser: function() {
            var webkitVer = parseInt((/WebKit\/([0-9]+)/.exec(navigator.appVersion) || 0)[1],10) || void 0;
            if (webkitVer === undefined) {
                return false;
            }
            return webkitVer <= 536;
        }
´´´

**Comment by patrickkettner** [comment]
Hey @stnor!

Which versions of anrdoi/iPad/iPhone have a webkit versions below 537

**Comment by stnor** [comment]
Android 4.1.x (might also be Android 4.2 (not sure)
IOS  6.1.3 and below

**Comment by stnor** [comment]
http://jimbergman.net/webkit-version-in-android-version/

**Comment by stnor** [comment]
I am not sure if the issue is in Android 4.2 or higher. I got a report on Mozilla/5.0 (Linux; U; Android 4.1.1; sv-se; PadFone 2 Build/JRO03L) AppleWebKit/534.30 at least.

**Comment by patrickkettner** [comment]
I haven't been able to test this myself, but [caniuse](http://caniuse.com/#feat=css-transitions) seems to disagree with what you are saying.

Could you supply a code example showing the issue?

**Comment by patrickkettner** [comment]
are you certain you are using vendor prefixes (`-webkit-`?)

**Comment by stnor** [comment]
if (!this.isBrokenWebkitBrowser() && Modernizr.csstransitions) {
                    previousView.$el.one('transitionend webkitTransitionEnd oTransitionEnd otransitionend MSTransitionEnd',
                    _.bind(function() {
                        // Some browsers fire more than one of these...
                        if (previousView.shown !== true) {
                            previousView.shown = true;
                            previousView.destroy();
                            this.renderQuiz();
                        }
                    }, this));
                    $("#question_container").css("opacity", 0);
                } else {
                    previousView.destroy();
                    this.renderQuiz();
                }

**Comment by patrickkettner** [comment]
@stnor - I wasn't clear, sorry about that!

I am asking for a page that has css transitions on it that is broken on those versions, where modernizr say it is not broken

**Comment by stnor** [comment]
Transitionend (none of the) doesn't seem to fire in these browsers

**Comment by stnor** [comment]
http://127.0.0.1:8080/nomp/#!/math/21/multiplication/two-times-tables/21.MULTIPLICATION.2

**Comment by stnor** [comment]
The user doesn't get to the second question, as the transition end (events) never fires.

**Comment by patrickkettner** [comment]
@stnor theres a _lot_ of code there. just a quick grep and I don't see any I won't be able to dig into a full site for a while, but if you send a reduced test case (ie a page with just modernizr and a css transition) I can get it fixed right off.

**Comment by stnor** [comment]
I'll give it a go.

**Comment by stnor** [comment]
Try this:
http://jsfiddle.net/5wocynq6/

**Comment by stnor** [comment]
There. sorry. http://jsfiddle.net/5wocynq6/4/

**Comment by stnor** [comment]
http://jsfiddle.net/5wocynq6/5/

I might have screwed up... Perhaps i need the prefixes in transition for it to work in those browsers...

**Comment by patrickkettner** [comment]
yep! that would be an issue

**Comment by stnor** [comment]
Thanks (sorry)

**Comment by patrickkettner** [comment]
no worries!

On Mon, Jan 26, 2015 at 4:35 PM, Stefan Norberg <notifications@github.com>
wrote:

> Closed #1501 <https://github.com/Modernizr/Modernizr/issues/1501>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Modernizr/Modernizr/issues/1501#event-224940706>.
>



-- 
patrick
